source/shared/cpp/ObjectModel/TextBlock.cpp

```diff
@@ -165,3 +168,141 @@ void TextBlock::SetHorizontalAlignment(const HorizontalAlignment value)
     m_hAlignment = value;
 }
 
+std::string TextBlock::ParseISO8601UsingRegex() const
+{
+	std::regex pattern("\\{\\{(DATE|Date|TIME|Time)\\((\\d{4})-{1}(\\d{2})-{1}(\\d{2})T(\\d{2}):{1}(\\d{2}):{1}(\\d{2})(Z|([+-])(\\d{2}):{1}(\\d{2}))(\\)|,Short\\)|,SHORT\\)|,Long\\)|,LONG\\))\\}\\}");
+    std::smatch matches;
+    std::string text = m_text;
+    std::ostringstream parsedostr;
+    time_t offset = 0; 
+    bool isDate = false, isShort = true;
+    enum MatchIndex
+    {
+        IsDate = 1,
+        YEAR,
+        Month,
+        Day,
+        Hour,
+        Min,
+        Sec,
+        TimeZone,
+        TZHr = 10,
+        TZMn,
+        Format,
+    };
+    int factor = 1, hours = 0, minutes = 0;
+    struct tm parsedTm = { 0 };
+    int *addrs[] = {&parsedTm.tm_year, &parsedTm.tm_mon, 
```

Comment:
>  [](start = 56, length = 1)

[nit] Please check for trailing spaces throughout this file

Reply:
thanks; i need to import vim settings from mac.
it kept get lost after windows update

---
In reply to: [149179442](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149179442) [](ancestors = 149179442)